File: data/CodeLlama/readable_format_HPEC/fortran/file153.f90
subroutine l2norm ( ldx, ldy, ldz,  &
&                    nx0, ny0, nz0,  &
&                    ist, iend,  &
&                    jst, jend,  &
&                    v, sum )


implicit none

integer ldx, ldy, ldz
integer nx0, ny0, nz0
integer ist, iend
integer jst, jend
double precision  v(5,ldx/2*2+1,ldy/2*2+1,*), sum(5)

integer i, j, k, m


do m = 1, 5
sum(m) = 0.0d+00
end do

do k = 2, nz0-1
do j = jst, jend
do i = ist, iend
do m = 1, 5
sum(m) = sum(m) + v(m,i,j,k)*v(m,i,j,k)
end do
end do
end do
end do

do m = 1, 5
sum(m) = sqrt ( sum(m) / ( dble(nx0-2)*(ny0-2)*(nz0-2) ) )
end do

return
end

File: data/llama3_8B_ins_results/readable_format_HPEC/cpp/file153.cpp
#include <iostream>
#include <cmath>

void l2norm(int ldx, int ldy, int ldz, int nx0, int ny0, int nz0, int ist, int iend, int jst, int jend, double v[5][ldx/2*2+1][ldy/2*2+1][3], double sum[5]) {
    for (int m = 0; m < 5; m++) {
        sum[m] = 0.0;
    }

    for (int k = 1; k < nz0; k++) {
        for (int j = jst; j <= jend; j++) {
            for (int i = ist; i <= iend; i++) {
                for (int m = 0; m < 5; m++) {
                    sum[m] += v[m][i][j][k] * v[m][i][j][k];
                }
            }
        }
    }

    for (int m = 0; m < 5; m++) {
        sum[m] = std::sqrt(sum[m] / ((double)(nx0-2) * (ny0-2) * (nz0-2)));
    }
}
